openrr-planner/src/collision/self_collision_checker.rs

```diff
@@ -7,18 +7,20 @@ use serde::{Deserialize, Serialize};
 use tracing::debug;
 
 use crate::{
-    collision::parse_colon_separated_pairs, errors::*, interpolate, CollisionDetector,
-    TrajectoryPoint,
+    collision::{parse_colon_separated_pairs, RobotCollisionDetector},
+    errors::*,
+    interpolate, CollisionDetector, TrajectoryPoint,
 };
 
 pub struct SelfCollisionChecker<N>
 where
     N: RealField + Copy + k::SubsetOf<f64>,
 {
-    pub collision_check_robot: Arc<k::Chain<N>>,
-    pub collision_detector: CollisionDetector<N>,
-    pub collision_pairs: Vec<(String, String)>,
-
+    /// Robot for reference (read only and assumed to hold the latest full states)
+    pub reference_robot: Arc<k::Chain<N>>,
+    /// Robot collision detector
+    pub robot_collision_detector: RobotCollisionDetector<N>,
```

Comment:
Just comment:

I used to be aggressive, so all members of this structure are public, but I think it is basically better to keep it private if possible.